crates/fuel-core/src/executor.rs

```diff
@@ -625,36 +611,47 @@ where
 
         // After the execution of all transactions in production mode, we can set the final fee.
         if execution_kind == ExecutionKind::Production {
-            coinbase_tx.outputs_mut().clear();
-            coinbase_tx.outputs_mut().push(Output::coin(
-                self.config.coinbase_recipient,
-                execution_data.coinbase,
-                *self.config.consensus_parameters.base_asset_id(),
-            ));
-            block.transactions[0] = coinbase_tx.clone().into();
-        }
+            let amount_to_mint = if self.config.coinbase_recipient != ContractId::zeroed()
+            {
+                execution_data.coinbase
+            } else {
+                0
+            };
 
-        if execution_kind != ExecutionKind::DryRun {
-            coinbase_tx = self.check_coinbase(
-                block_height,
-                coinbase_tx,
-                Some(execution_data.coinbase),
-            )?;
-            self.apply_coinbase(
-                coinbase_tx,
-                block,
+            let coinbase_tx = Transaction::mint(
+                TxPointer::new(block_height, execution_data.tx_count),
+                input::contract::Contract {
+                    utxo_id: Default::default(),
```

Comment:
It seems weird to be using `Default` for these values. If the values are static, maybe have the constructor include them under the hood? But I have a feeling we should be explicit about what these values are so if the defaults change from under us we aren't surprised, etc.